x-pack/legacy/plugins/siem/server/lib/detection_engine/routes/rules/import_rules_route.test.ts

```diff
@@ -276,114 +208,79 @@ describe('import_rules_route', () => {
-        });
-        expect(statusCode).toEqual(200);
-
-        clients.alertsClient.find.mockResolvedValue(getFindResultWithSingleHit());
-        clients.alertsClient.get.mockResolvedValue(getResult());
-
-        const { statusCode: statusCodeRequest2, payload: payloadRequest2 } = await server.inject(
-          getImportRulesRequestOverwriteTrue(requestPayload)
+        clients.alertsClient.find.mockResolvedValue(getFindResultWithSingleHit()); // extant rule
+        const overwriteRequest = getImportRulesRequestOverwriteTrue(
+          buildHapiStream(ruleIdsToNdJsonString(['rule-1']))
         );
-        const parsedRequest2: ImportSuccessError = JSON.parse(payloadRequest2);
+        const response = await server.inject(overwriteRequest, context);
 
-        expect(parsedRequest2).toEqual({
+        expect(response.status).toEqual(200);
+        expect(response.body).toEqual({
           errors: [],
           success: true,
           success_count: 1,
         });
-        expect(statusCodeRequest2).toEqual(200);
       });
     });
   });
 
   describe('multi rule import', () => {
     test('returns 200 if all rules imported successfully', async () => {
-      clients.alertsClient.find.mockResolvedValue(getFindResult());
-
-      const requestPayload = getSimpleRuleAsMultipartContent(['rule-1', 'rule-2']);
-      const { statusCode, payload } = await server.inject(getImportRulesRequest(requestPayload));
-      const parsed: ImportSuccessError = JSON.parse(payload);
+      const multiRequest = getImportRulesRequest(
+        buildHapiStream(ruleIdsToNdJsonString(['rule-1', 'rule-2']))
+      );
+      const response = await server.inject(multiRequest, context);
 
-      expect(parsed).toEqual({
+      expect(response.status).toEqual(200);
+      expect(response.body).toEqual({
         errors: [],
         success: true,
         success_count: 2,
       });
-      expect(statusCode).toEqual(200);
     });
 
     test('returns 200 with errors if all rules are missing rule_ids and import fails on validation', async () => {
-      clients.alertsClient.find.mockResolvedValue(getFindResult());
-
-      const requestPayload = getSimpleRuleAsMultipartContentNoRuleId(2);
-      const { statusCode, payload } = await server.inject(getImportRulesRequest(requestPayload));
-      const parsed: ImportSuccessError = JSON.parse(payload);
+      const rulesWithoutRuleIds = ['rule-1', 'rule-2'].map(ruleId => getSimpleRuleWithId(ruleId));
+      const badPayload = buildHapiStream(rulesToNdJsonString(rulesWithoutRuleIds));
+      const badRequest = getImportRulesRequest(badPayload);
 
-      expect(parsed.success).toEqual(false);
-      expect(parsed.errors[0].error.message).toEqual(
-        'child "rule_id" fails because ["rule_id" is required]'
-      );
-      expect(parsed.errors[0].error.status_code).toEqual(400);
-      expect(statusCode).toEqual(200);
-    });
-
-    test('returns 200 with reported conflict if error parsing rule', async () => {
```

Comment:
@yctercero I think I removed this test because it was redundant with a simpler validation test, but I'd appreciate if you gave this whole file a once-over.

Reply:
Looked through all the tests, looks great! Thanks for doing this. Went through and it looks like we hit all the various branchings.